website/static/js/registrationRetraction.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * On page load, maintains knockout ViewModel
+**/
+'use strict';
+
+var ko = require('knockout');
+require('knockout.validation');
+var $ = require('jquery');
+var Raven = require('raven-js');
+
+var $osf = require('js/osfHelpers');
+var koHelpers = require('js/koHelpers');
+
+var RegistrationRetractionViewModel = function(submitUrl, registrationTitle) {
+
+    var self = this;
+
+    self.registrationTitle = registrationTitle;
+    self.justification = ko.observable('').extend({
+        maxLength: 2048
+    });
+    self.confirmationText = ko.observable().extend({
+        required: true,
+        mustEqual : self.registrationTitle
+    });
+    self.onSubmitSuccess = function(response) {
+        window.location = response.redirectUrl;
+    };
+    self.onSubmitError = function(error) {
```

Comment:
The signature of this error handler is incorrect. Failure callbacks have the signature `jqxhr.fail(function(xhr, status, errorThrown)`